core/src/main/java/io/kestra/core/docs/ClassPluginDocumentation.java

```diff
@@ -125,7 +125,13 @@ private static Map<String, Object> flatten(Map<String, Object> map, List<String>
 
     @SuppressWarnings("unchecked")
     private static Map<String, Object> flatten(Map<String, Object> map, List<String> required, String parentName) {
-        Map<String, Object> result = new TreeMap<>();
+        Map<String, Object> result = new TreeMap<>((key1, key2) -> {
+            boolean key1Required = required.contains(key1);
+            boolean key2Required = required.contains(key2);
+            if(key1Required == key2Required) return key1.compareTo(key2);
```

Comment:
As it's boolean it's more logical to use `&&` instead of `==` but you can keep it if you want.